src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -573,33 +579,38 @@ public final String readUTF() throws IOException {
      */
     public static final String readUTF(DataInput in) throws IOException {
         int utflen = in.readUnsignedShort();
-        byte[] bytearr = null;
-        char[] chararr = null;
-        if (in instanceof DataInputStream dis) {
+        byte[] bytearr;
+        DataInputStream dis = null;
+        if (in instanceof DataInputStream) {
+            dis = (DataInputStream)in;
```

Comment:
I guess that not making use of `instanceof` pattern matching is to enable backporting before JDK 16?

Reply:
I have the same question here.